source/custom/nixnet_converters.h

```diff
@@ -0,0 +1,171 @@
+#ifndef NIDEVICE_GRPC_DEVICE_NIXNET_CONVERTERS_H
+#define NIDEVICE_GRPC_DEVICE_NIXNET_CONVERTERS_H
+#include <nixnet.h>
+#include <nixnet.pb.h>
+
+// Add underscore to usings so they don't conflict with including files in the same namespace.
+namespace pb_ = ::google::protobuf;
+
+namespace nixnet_grpc {
+
+struct FrameHolder {
+  FrameHolder(const pb_::RepeatedPtrField<FrameBuffer>& input)
+  {
+    // all frames in repeated field must be of same type
+    switch (input[0].frame_case()) {
+      case FrameBuffer::kCan:
```

Comment:
Why is it better to have a separate frame type for every one of these? Could the types just be:

```proto
enum FrameType { Frame, EnetFrame }
```

Both the underlying type and the proto type are the same. And it doesn't look like we validate that the actual *data* is a frame (could we do that by looking at the type?).

I think this implementation is OK, though. It makes a little more sense to me with just 2 frame types, but I'm not sure I understand all the considerations. 

Pros/Cons: 
* Using only 2 frame types would make it "just work" with a new frame type.
* Having a type struct for each subtype allows us to change them independently in grpc (this would be an important consideration in normal proto design. But, in this case, I'm not sure if the underlying C API representation supports that or not).
* In this implementation, is it possible for a user to request "LIN" data and then have us return an array of "LIN" frames that are actually "CAN" frames? Are we implying type-checking that we're not actually doing?

Reply:
I had initially thought of using Frame for LIN/CAN/FlexRay and J1939, since underlying data structure is same (difference is mostly in payload lengths, and some connotation about how `PayloadLength` gets calculated. But these differences are hidden away by macros from xnet driver.

Main reason for using separate types is mostly to do with discoverability in request/response messages. I couldn't find a single name good enough to imply that use this field to pass in either of those 4 frame types. Other name `EnetFrame` stands on its own and implies the type very well. I will discuss this with Waldemar post this PR to see if there is single name that domain users would understand. Maybe just `Frame` could be good enough too.

> In this implementation, is it possible for a user to request "LIN" data and then have us return an array of "LIN" frames that are actually "CAN" frames? Are we implying type-checking that we're not actually doing?

Yes, this is possible. We don't do any type checking of any sort on server side. I am not sure if this is completely possible either. There are few FrameType constants of each type in header file (and also few more frame types which are common for all frames). But if we could merge the frames into single field as you suggested, that would solve this problem too.

Reply:
> I will discuss this with Waldemar post this PR to see if there is single name that domain users would understand. Maybe just Frame could be good enough too.

That sounds good. Whatever you decide is OK with me (including the current implementation). And you don't need to block this PR on the decision if you don't want to.

Reply:
I discussed this with Waldemar and he mentioned having separate fields for each frame type (even though underlying struct type is same) is better. There is no single name we could use to refer to all those 4 types of frames.

Regarding validation if the frames being returned are of same type that user has requested, Waldemar said that is not required. Type of frame depends on how session is initialized. In C API, users can initialize LIN session and pass CAN frame to write API, and that doesn't throw any error either. Users might see weird output though. If users try to pass J1939 frame for LIN session, that might throw error because J1939 supports longer payload than LIN. As long as we are reporting such errors from driver back to users (which we do), this should be sufficient.